### PR TITLE
[9.x] Make `whereBelongsTo` accept Collection

### PR DESCRIPTION
**Pull request by erikgaal**
Hi there!

Extending upon #38927, I'm submitting this PR that allows you to retrieve records using a collection of related records. 

Previously, if you wanted to retrieve, for example, all Posts that belong to one of multiple Categories, you would need to perform the following query:

```php
$query->whereBelongsTo($category[0])
    ->orWhereBelongsTo($category[1])
    // etc...

// Or:
$query->whereIn('category_id', $categories->modelKeys());
```

Same as in the aforementioned PR, you would need to know the key names when building and maintain these if they change. This PR adds support for Collections passed to the `whereBelongsTo` method:

```php
$query->whereBelongsTo($categories);
```

```php
$query->whereBelongsTo($categories, 'category');
```

### CODE REVIEW

**Comment by a-bashtannik** [comment]
Could somebody explain or point to the docs, why [whereIn](https://github.com/laravel/framework/blob/b9a34d01bf19347aef3f04b11709f1ca8c9deddf/src/Illuminate/Database/Query/Builder.php#L1001) accepts a flag `boolean` supposing 2 values: `and` and `or` while the SQL IN operator is a shorthand for multiple `or` conditions? Thanks!

**Comment by erikgaal** [comment]
@a-bashtannik It allows you to make the distinction between adding `and "column" in  ?` and `OR "column" IN ?` to the SQL query. It's the same for all `where` methods in the Builder.

**Comment by a-bashtannik** [comment]
@erikgaal argghh so simple true, thanks!

**Comment by taylorotwell** [comment]
@erikgaal did you give any thoughts to just baking this functionality into `whereBelongsTo`?

**Comment by erikgaal** [comment]
> @erikgaal did you give any thoughts to just baking this functionality into `whereBelongsTo`? 

No, I haven't. Seems logical to support both a single `Model` and a `Collection` as argument to the function. If that works for you too, I'm happy to implement it that way!

**Comment by taylorotwell** [comment]
@erikgaal yeah I think it's fine to bake this behavior into the existing method.

**Comment by erikgaal** [comment]
@taylorotwell Done!

File: src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php [hunk 1]
```diff
@@ -470,6 +471,17 @@ public function orWhereMorphedTo($relation, $model)
      */
     public function whereBelongsTo($related, $relationshipName = null, $boolean = 'and')
     {
+        if (! $related instanceof Collection) {
+            $values = $related->newCollection([$related]);
+        } else {
+            $values = $related;
+            $related = $values->first();
```
Comment (Jubeki) [inline src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php:478]: What happens if you pass in an empty collection? I would expect to get no models back after the query is run.

**Comment by Bosphoramus** [comment]
Hi @erikgaal would it be possible for empty collections to be passed?

Id expect it to work the same as if i pass [] to whereIn(), no records would be returned, this would be handy to avoid some conditional checks.

**Pull request merged**